src/app/site/[site]/sitemap.xml/route.ts

```diff
@@ -0,0 +1,58 @@
+import dayjs from "dayjs"
+import { NextRequest } from "next/server"
+
+import { QueryClient } from "@tanstack/react-query"
+
+import { getSiteLink } from "~/lib/helpers"
+import { NextServerResponse } from "~/lib/server-helper"
+import { PageVisibilityEnum } from "~/lib/types"
+import { fetchGetPagesBySite } from "~/queries/page.server"
+import { fetchGetSite } from "~/queries/site.server"
+
+export const GET = async (req: NextRequest) => {
+  const response = new NextServerResponse()
+
+  const queryClient = new QueryClient()
+  // http://localhost:2222/site/innei-4525/sitemap.xml
+  // http://innei-4525.localhost:2222/sitemap.xml
+  // find innei-5425 site handle by regexp
+  const domainOrSubdomain =
+    req.nextUrl.pathname.match(/\/site\/(.*)\/sitemap.xml/)?.[1] ||
+    // http://innei-4525.localhost:2222/sitemap.xml
+    req.headers.get("host")?.split(".")[0] ||
+    ""
```

Comment:
It seems that the custom domain is not functioning properly. I suggest adding a header `x-handle` to the middleware https://github.com/Crossbell-Box/xLog/blob/dev/src/middleware.ts#L86 and retrieving the handle from the header.

Reply:
I tested `http://innei-4525.localhost:2222/sitemap.xml` before and there was no problem. Does it refer to other meanings? I don't understand.

Reply:
I am referring to custom domains like `https://blog.innei.ren/`.